discovery_engine/lib/src/domain/repository/sources_repo.dart

```diff
@@ -0,0 +1,22 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+/// Repository interface for source domains excluded from the feed.
+abstract class SourcesRepository {
+  /// Get a set of all favourite sources.
+  Future<Set<String>> getAll();
+
+  /// Persist a set of favourite sources.
+  Future<void> save(Set<String> sources);
+}
```

Comment:
Will this be merged with the `ExcludedSourcesRepository` in a later PR.

If so maybe we should call the methods `getFavorites`/`saveFavorites` or similar.

If not this probably should still be called `FavoritSourceRepository`

Reply:
let's keep them separate for now. whoever ends up working on the dart side of this story can later refactor this if necessary - I only introduced this here to have the minimum amount of dart in place for the changes on the rust side to go through.

btw you didn't prefix your comments with PR-emojis :wink:

Reply:
see [later comment](https://github.com/xaynetwork/xayn_discovery_engine/pull/283#discussion_r839719445)